### ISSUE TITLE
AppSidebar tabs are not updated

### ISSUE THREAD

**Issue by nickvergessen** [nextcloud/nextcloud-vue#747]
1. Have a `AppSidebar` with 3 `AppSidebarTab` inside.
2. On one of the items have a condition: ` v-if="showChatInSidebar"`
3. Trigger something that changes the value of this variable

### Expected

Chat tab should now show

### Actually

Chat tab does not show.

When you close the sidebar and reopen it, the tab will be shown.
From my understanding calling `updateTabs` only on `mounted` is the problem:
https://github.com/nextcloud/nextcloud-vue/commit/6838034e0da49c3ae495c9b12fcb5c3fd6ff1a86#diff-8f46855f7ea398f0fef41785098762daR171

Closing/reopening the sidebar removes the Sidebar and recreates it, so it is "re" mounted and the list updates. But that is not a solution. Is it possible to "watch" the `this.$children` somehow and also update in this case?

**Comment by skjnldsv** [comment]
You can use v-show and it should work, right?
Children is not reactive unfortunately, we had to change the behaviour of the actions fro the same reason iirc :thinking:

**Comment by nickvergessen** [comment]
No, v-show will always show the item, because it's there just hidden with css, but since you loop over the children, it is there.

**Comment by skjnldsv** [comment]
yes, that's why I'm saying, don't use v-if, use v-show.

**Comment by nickvergessen** [comment]
I tried it before doing the previous comment already and it doesnt work:

* with `v-if` it will not be added to `AppSidebar.tabs` when the condition turns true
* with `v-show` it will always be shown even when the condition is false

**Comment by skjnldsv** [comment]
> * with `v-show` it will always be shown even when the condition is false

I highly suspect a css rule behind this. Give me some reproducible code pls.
Anyway, let's see what we can do.

**Comment by nickvergessen** [comment]
Code is at:
https://github.com/nextcloud/spreed/pull/2486/files

Play around with
https://github.com/nextcloud/spreed/pull/2486/files#diff-ea994a51b0d9884f4fc046eca11a5c99R91
and
https://github.com/nextcloud/spreed/pull/2486/files#diff-8067ecb062a14f890987cbca45e2168eR48
to have the fun

**Comment by dartcafe** [comment]
Same behavior here. v-if condition of AppSidebarTab is not reactive 
https://github.com/nextcloud/polls/blob/master/src/js/components/SideBar/SideBar.vue

**Comment by nickvergessen** [comment]
Bump
